### PR TITLE
Define when changes to session state are applied.

### PR DESCRIPTION
**Pull request by toji**
Makes the frame lifetime more explicit and explains what session data is captured and when. This is not something that's had an explicit issue created for it, but it has come up repeatedly as we've discussed various mutable bits of session state. Will become even more important as we add more mutable state to the session in the future.

Also contemplating a related change to make the state that changes on a per-frame basis more explicit, but that'll be a followup to this.

### CODE REVIEW

File: explainer.md [hunk 1]
```diff
@@ -251,6 +255,10 @@ function drawScene(view) {
 }
 ```
 
+### Frame lifetime
```
[the remark this answers is not in the document]
Comment (toji) [reply explainer.md]: Derp. Was considering breaking off into it's own section then reconsidered. Guess I left a piece behind.

File: explainer.md [hunk 2]
```diff
@@ -197,10 +197,14 @@ If the system's underlying XR device changes (signaled by the `devicechange` eve
 
 The WebXR Device API provides information about the current frame to be rendered via the `XRFrame` object which developers must examine each iteration of the render loop. From this object the frame's `XRViewerPose` can be queried, which contains the information about all the views which must be rendered in order for the scene to display correctly on the XR device.
 
-`XRWebGLLayer` objects are not updated automatically. To present new frames, developers must use `XRSession`'s `requestAnimationFrame()` method. When the callback function is run, it is passed both a timestamp and an `XRFrame` containing fresh rendering data that must be used to draw into the `XRWebGLLayer`s `framebuffer` during the callback.
+`XRWebGLLayer` objects are not updated automatically. To present new frames, developers must use `XRSession`'s `requestAnimationFrame()` method. The first time `requestAnimationFrame()` is called for a session a new `XRFrame` is created internally. Subsequent calls to `requestAnimationFrame()` will all use the same `XRFrame` until the frame callbacks are run. At that point the next `requestAnimationFrame()` call will create a new `XRFrame` and the cycle repeats.
```
[the remark this answers is not in the document]
Comment (toji) [reply explainer.md]: Worked on the wording a bit more, and used the phrasing you suggested in particular. Let me know how it reads to you now.

File: explainer.md [hunk 3]
```diff
@@ -197,9 +197,11 @@ If the system's underlying XR device changes (signaled by the `devicechange` eve
...
 
-`XRFrame` objects act as snapshots of the state of the XR device and all associated inputs. The state may represent historical data, current sensor readings, or a future projection. Due to it's time-sensitive nature, an `XRFrame` is only valid during the execution of the callback that it is passed into, and once control is returned to the browser any active `XRFrame` objects are marked as inactive. Calling any method of an inactive `XRFrame` will throw a [`NotAllowedError`](https://heycam.github.io/webidl/#notallowederror).
+A new `XRFrame` is created for each batch of `requestAnimationFrame()` callbacks. `XRFrame` objects act as snapshots of the state of the XR device and all associated inputs. The state may represent historical data, current sensor readings, or a future projection. Due to it's time-sensitive nature, an `XRFrame` is only valid during the execution of the callback that it is passed into, and once control is returned to the browser any active `XRFrame` objects are marked as inactive. Calling any method of an inactive `XRFrame` will throw a [`NotAllowedError`](https://heycam.github.io/webidl/#notallowederror).
```
Comment (thetuvix) [inline explainer.md]: The first sentence here implies that `XRFrame`s are basically created only for rAF callbacks, while another key source is input event callbacks.  This is key context for the third sentence, which talks about "historical data", as the rAF `XRFrame` instances will all be about a future-predicted photon time.

Comment (toji) [reply explainer.md]: Good catch. Updated that line to say:

> A new `XRFrame` is created for each batch of `requestAnimationFrame()` callbacks or for certain events that are associated with tracking data.

**Comment by toji** [comment]
Thanks for the feedback @thetuvix! At this point since it doesn't seem like there's any further feedback I'll go ahead and merge this change in when @NellWaliczek clears her review bit.

File: explainer.md [hunk 4]
```diff
@@ -197,9 +197,11 @@ If the system's underlying XR device changes (signaled by the `devicechange` eve
...
+A new `XRFrame` is created for each batch of `requestAnimationFrame()` callbacks or for certain events that are associated with tracking data. `XRFrame` objects act as snapshots of the state of the XR device and all associated inputs. The state may represent historical data, current sensor readings, or a future projection. Due to it's time-sensitive nature, an `XRFrame` is only valid during the execution of the callback that it is passed into, and once control is returned to the browser any active `XRFrame` objects are marked as inactive. Calling any method of an inactive `XRFrame` will throw a [`NotAllowedError`](https://heycam.github.io/webidl/#notallowederror).
+
+The `XRFrame` also makes a copy of the  `XRSession`'s "render state", such as `depthNear/Far` values and the `baseLayer`, at the time the first `requestAnimationFrame()` call in the current batch was made. This captured render state is what will be used when the frame is being composited by the XR hardware. Any subsequent changes the developer makes to the session's render state will not be applied until the the next `XRFrame` is created.
```
[the remark this answers is not in the document]
Comment (toji) [reply explainer.md]: Good call. Updated to say:

>  This captured render state is what will be used **when computing view information like projection matrices and** when the frame is being composited by the XR hardware.

**Pull request merged**